flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/model/node/NewConnection.java

```diff
@@ -79,11 +79,19 @@ public Map<Path, List<TextEdit>> toSource(SourceBuilder sourceBuilder) {
         if (scope.isEmpty()) {
             throw new IllegalStateException("Scope is not defined for the new connection node");
         }
-        return switch (scope.get().value().toString()) {
-            case Property.LOCAL_SCOPE -> sourceBuilder.textEdit(false).build();
-            case Property.GLOBAL_SCOPE -> sourceBuilder.textEdit(false, "connections.bal", true).build();
+        switch (scope.get().value().toString()) {
+            case Property.LOCAL_SCOPE -> {
+                Path file = sourceBuilder.filePath.getFileName();
+                if (file == null) {
+                    throw new IllegalStateException("File is not defined for the new connection node");
+                }
+                return sourceBuilder.textEdit(false, file.toString(), true).build();
+            }
+            case Property.GLOBAL_SCOPE -> {
+                return sourceBuilder.textEdit(false, "connections.bal", true).build();
+            }
             default -> throw new IllegalStateException("Invalid scope for the new connection node");
```

Comment:
Adding `acceptImport` is sufficient

```suggestion
        return switch (scope.get().value().toString()) {
            case Property.LOCAL_SCOPE -> sourceBuilder.textEdit(false).acceptImport().build();
            case Property.GLOBAL_SCOPE -> sourceBuilder.textEdit(false, "connections.bal", true).build();
            default -> throw new IllegalStateException("Invalid scope for the new connection node");
        };
```